lib/engine/stock_market.rb

```diff
@@ -12,11 +12,18 @@ def initialize(market, unlimited_types, multiple_buy_types: [], zigzag: nil)
       @zigzag = zigzag
       @market = market.map.with_index do |row, r_index|
         row.map.with_index do |code, c_index|
-          price = SharePrice.from_code(code,
-                                       r_index,
-                                       c_index,
-                                       unlimited_types,
-                                       multiple_buy_types: multiple_buy_types)
+          price = if code.instance_of?(Hash)
+                    SharePrice.new([r_index, c_index],
+                                   price: code[:price],
+                                   types: code[:types],
+                                   info: code[:info])
```

Comment:
Don't forget unlimited_types and multiple_buy_types:

Reply:
Fixed